### ISSUE TITLE
Destroy triggered after matching terraform import. 

### ISSUE THREAD

**Issue by duxing** [hashicorp/terraform#15739]
### Description
Trying to import an existing elasticache cluster into terraform. Did import and created a tf file that matches the configuration in tfstate. But running terraform plan shows a destroy and a create.
According to this:
https://www.terraform.io/docs/providers/aws/r/elasticache_cluster.html
other than count, there's no way to specify the details on the nodes inside the cluster. I wonder if that's the reason why this is triggering a destroy. 

### Terraform Version
0.9.11

### Terraform Configuration Files
```hcl
resource "aws_elasticache_cluster" "elasticache" {
  availability_zone = "us-west-2a"
  cluster_id = "terraform-import"
  engine = "memcached"
  engine_version = "1.4.34"
  node_type = "cache.t2.micro"
  port = "11211"
  maintenance_window = "tue:12:00-tue:13:00"
  num_cache_nodes = "1"

  parameter_group_name = "default.memcached1.4"
  security_group_ids = ["sg-b82852dd", "sg-b87d9fdc"]
  subnet_group_name = "subnet-test"
}
```

### Debug Output
+ aws_elasticache_cluster.elasticache
    apply_immediately:            "<computed>"
    availability_zone:            "us-west-2a"
    az_mode:                      "<computed>"
    cache_nodes.#:                "<computed>"
    cluster_address:              "<computed>"
    cluster_id:                   "terraform-import"
    configuration_endpoint:       "<computed>"
    engine:                       "memcached"
    engine_version:               "1.4.34"
    maintenance_window:           "tue:12:00-tue:13:00"
    node_type:                    "cache.t2.micro"
    num_cache_nodes:              "1"
    parameter_group_name:         "default.memcached1.4"
    port:                         "11211"
    replication_group_id:         "<computed>"
    security_group_ids.#:         "2"
    security_group_ids.605997936: "sg-b87d9fdc"
    security_group_ids.909628542: "sg-b82852dd"
    security_group_names.#:       "<computed>"
    snapshot_window:              "<computed>"
    subnet_group_name:            "subnet-test"

- aws_elasticache_cluster.import-test

### Expected Behavior
Terraform plan output shows no change needed.

### Actual Behavior
1 '+' and 1 '-':
the existing resource is destroyed and a new one with identical configuration is stood up.

### Steps to Reproduce
1. create an elasticache cluster via AWS console. If it matters, use memcached 1.4.34. everything else use the default setting. 
2. run 'terraform import aws_elasticache_cluster.<your_resource_name> <your_elasticache_cluster_id>' to import the tfstate file.
3. create a new tf file with matching configurations. Something similar to what I uploaded.
4. run 'terraform plan' and observe.

### Important Factoids
Nothing special

**Comment by tombuildsstuff** [comment]
Hi @duxing 

Thanks for opening this issue :)

So as part of importing a resource, the Resource Name to import it as is specified as an command line argument:
```
$ terraform import aws_elasticache_cluster.import-test my_cluster
```

The Resource Name (`import-test` in the example above) which is specified on the command line needs to match the Resource Name in the Terraform configuration file. From what I can see, you're seeing this issue because the resource has been imported into the Statefile with one Resource Name, and is stored in the Terraform configuration as another. Given these identifiers don't match - Terraform thinks it needs to recreate this - as such this can be fixed by either:

1) Updating the Resource Name in the Terraform configuration to be named `import-test` . By which I mean changing:
```
resource "aws_elasticache_cluster" "elasticache" {
```
for
```
resource "aws_elasticache_cluster" "import-test" {
```

2) Moving the resource in the state to match the Terraform Configuration file i.e. `terraform state mv aws_elasticache_cluster.import-test aws_elasticache_cluster.elasticache`

Would you be able to take a look and confirm if that solves your issue?

Thanks!

**Comment by duxing** [comment]
Thanks for explaining. I've tested again by modifying the resource name and the output matches what I've expected.

I've built this test case because one of my elasticache cluster failed being terraformed with the same symptom(but I did use the right resource name). I copy-pasted the tf file for testing but forgot to change the name. 

Let me dig into the original case a bit more and possibly ask you with a further question.
Thanks again!
